lib/ansible/plugins/become/runas.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    become: runas
+    short_description: Run As user
+    description:
+        - This become plugins allows your remote/login user to execute commands as another user via the windows runas facility.
+    author: ansible (@core)
+    version_added: "2.8"
+    options:
+        become_user:
+            description: User you 'become' to execute the task
+            ini:
+              - section: privilege_escalation
+                key: become_user
+              - section: runas_become_plugin
+                key: user
+            vars:
+              - name: ansible_become_user
+              - name: ansible_runas_user
+            env:
+              - name: ANSIBLE_BECOME_USER
+              - name: ANSIBLE_RUNAS_USER
+            required: True
+        become_exe:
+            description: Sudo executable
+            default: runas
+            ini:
+              - section: privilege_escalation
+                key: become_exe
+              - section: runas_become_plugin
+                key: executable
+            vars:
+              - name: ansible_become_exe
+              - name: ansible_runas_exe
+            env:
+              - name: ANSIBLE_BECOME_EXE
+              - name: ANSIBLE_RUNAS_EXE
+        become_flags:
```

Comment:
While not necessary right now it would be nice if become_flags could take in a dict, or convert key=value, to a dict. Right now become_flags in runas is based on specific key=values and having the controller do the validation vs the remote execution would be nice.